web/__tests__/pages/api/post/post.int.test.ts

```diff
@@ -181,112 +183,101 @@ describe("[API] Post - Integration Test", () => {
         covered: false,
       });
       expect(uncoveredPosts).not.toBeNull();
       expect(uncoveredPosts.length).toBe(1);
+      expect(uncoveredPosts[0]._id).toMatchObject(uncoveredId);
     });
 
     test("random covered-uncovered", async () => {
-      let randomPost = createRandomPost();
-      const numPosts = Math.ceil(Math.random() * 10);
-      let numCovered = 0;
-      let numUncovered = 0;
+      const randPosts = randomPosts;
+      let allIds = [];
+      let covered = [];
+      let uncovered = [];
       let createPost;
-      for (let i = 0; i < numPosts; i++) {
-        const curr = Math.floor(Math.random() * 2);
-        if (curr == 0) {
-          randomPost.covered = false;
-          numUncovered++;
-        } else {
-          randomPost.covered = true;
-          numCovered++;
-        }
+      for (let i = 0; i < 10; i++) {
         const id = new mongoose.Types.ObjectId();
+        allIds.push(id.toString());
+        randPosts[i].covered
+          ? covered.push(id.toString())
+          : uncovered.push(id.toString());
         createPost = await Post.create({
-          ...randomPost,
+          ...randPosts[i],
           _id: id,
-          attachments,
         });
         expect(createPost).not.toBeNull();
       }
-      let goodWithArray = [];
-      for (const goodWithKey in goodWithMap) {
-        if (createPost[goodWithKey] == "yes") {
-          goodWithArray.push(goodWithMap[goodWithKey]);
-        }
-      }
-      const postFilterSchema = {
-        type: [randomPost.type],
-        breed: randomPost.breed,
-        age: [randomPost.age],
-        size: [randomPost.size],
-        gender: [randomPost.gender],
-        behavioral: randomPost.behavioral,
-        goodWith: goodWithArray,
+      const postFilters = {
+        type: Object.values(FosterType),
+        breed: Object.values(Breed),
+        age: Object.values(Age),
+        size: Object.values(Size),
+        gender: Object.values(Gender),
+        behavioral: Object.values(Behavioral),
+        goodWith: [],
       };
       const allPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
       });
       expect(allPosts).not.toBeNull();
-      expect(allPosts.length).toBe(numPosts);
+      expect(allPosts.length).toBe(randPosts.length);
+      let expected = allPosts.map(({ _id, ...user }) => _id.toString());
+      expect(expected).toEqual(expect.arrayContaining(allIds));
       const coveredPosts = await caller.post.getFilteredPosts({
-        postFilterSchema: postFilterSchema,
+        postFilterSchema: postFilters,
         covered: true,
       });
       expect(coveredPosts).not.toBeNull();
-      expect(coveredPosts.length).toBe(numCovered);
+      expect(coveredPosts.length).toBe(covered.length);
+      expected = coveredPosts.map(({ _id, ...user }) => _id.toString());
+      expect(expected).toEqual(expect.arrayContaining(covered));
```

Comment:
To be sure of the implementation, we need to check the entire array all fields within the objects.

```js
const expectedCoveredPosts = randomPosts.filter((post) => post.covered);
expect(coveredPosts).toMatchObject(expectedCoveredPosts);
```